store/mongo/migration_1_2_2.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package mongo
+
+import (
+	"strings"
+
+	"github.com/globalsign/mgo"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+)
+
+type migration_1_2_2 struct {
+	session *mgo.Session
+	db      string
+}
+
+// Up drops index with len(name) > 127 chars in the 'deployments' collection
+func (m *migration_1_2_2) Up(from migrate.Version) error {
+	s := m.session.Copy()
+	defer s.Close()
+
+	indicesArrays := [][]string{
+		StatusIndexes,
+		DeviceIDStatusIndexes,
+		DeploymentIdIndexes,
+	}
+	for _, index := range indicesArrays {
+		// DropIndex will use the same rules for exploding the index name
+		// as EnsureIndexKey previously used to create the 'long' index
+		err := s.DB(m.db).
+			C(CollectionDevices).
+			DropIndex(index...)
```

Comment:
do we need to drop old indexes? will there be some conflicts if we don't?

Reply:
> do we need to drop old indexes? will there be some conflicts if we don't?

There are two reasons:
 - `EnsureIndex` will not actually even connect to server once it returned with success, either `DropIndex` or `ResetIndexCache` has to be used to really be sure that `EnsureIndex`
 - it was a copy-paste from the way we deal with existing index in deployments

I do not think there are any conflicts that I can see.

Reply:
about `EnsureIndex`:
following requests **for the same index** will not contact the server
I think it's not the caseDropIndex

in the previous migration, `DropIndex` was used because we actually wanted to drop the index

Reply:
> about `EnsureIndex`:
> following requests **for the same index** will not contact the server
> I think it's not the case
> 
> in the previous migration, `DropIndex` was used because we actually wanted to drop the index

yes, of course, for the same index. it is only an extra precaution here.